test/e2e/metrics_adapter_test.go

```diff
@@ -58,6 +60,13 @@ func isAPIServiceAvailable(conditions []apiservicesv1.APIServiceCondition) bool
 	return false
 }
 
+func isAPIServicePointingToRightMetricsService(t *testing.T, metricsService *apiservicesv1.APIService) bool {
+	if f.IsFeatureGateEnabled(t, MetricsServerFeatureGate) {
+		return metricsService.Spec.Service.Name == "metrics-server"
+	}
+	return metricsService.Spec.Service.Name == "prometheus-adapter"
+}
+
 func TestMetricsAPIAvailability(t *testing.T) {
```

Comment:
I don't think we need those dedicatedServiceMonitors tests after https://github.com/openshift/cluster-monitoring-operator/pull/2160

I'll prepare a PR to clean that up to simplify the tests.

Reply:
here we go https://github.com/openshift/cluster-monitoring-operator/pull/2210